frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -88,72 +160,334 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+          setOperandsDeleteInProgress(false);
+          setOperandsDeleteFinished(true);
+        },
+      );
+    }
+    setOperandsDeleteFinished(!deleteOperands);
 
-    handlePromise(Promise.all(allPromises))
+    handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
       .then(() => {
-        close();
-        if (
-          window.location.pathname.split('/').includes(subscription.metadata.name) ||
-          window.location.pathname.split('/').includes(subscription.status.installedCSV)
-        ) {
-          history.push(resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()));
-        }
+        setOperatorUninstallFinished(true);
       })
-      .catch(() => {});
+      .catch(() => {
+        setOperatorUninstallFinished(true);
+      });
   };
 
   const name = csv?.spec?.displayName || subscription?.spec?.name;
+  const csvName = csv?.metadata?.name;
   const namespace =
     subscription.metadata.namespace === GLOBAL_OPERATOR_NAMESPACE
       ? 'all-namespaces'
       : subscription.metadata.namespace;
   const uninstallMessage = csv?.metadata?.annotations?.[OPERATOR_UNINSTALL_MESSAGE_ANNOTATION];
+
+  const Instructions = () => (
+    <>
+      <p>
+        <Trans t={t} ns="olm">
+          Operator <strong>{{ name }}</strong> will be removed from <strong>{{ namespace }}</strong>
+          . Click on the checkbox below to also remove all Operands associated with this Operator.
+          If your Operator configured off-cluster resources, these will continue to run and require
+          manual cleanup.
+        </Trans>
+      </p>
+      {removePlugins && (
+        <p>
+          {t('olm~The console plugin provided by this operator will be disabled and removed.', {
+            count: enabledPlugins.length,
+          })}
+        </p>
+      )}
+      {uninstallMessage && (
+        <>
+          <h2>{t('olm~Message from Operator developer')}</h2>
+          <p>{uninstallMessage}</p>
+        </>
+      )}
+    </>
+  );
+
+  const OperandsSection = () => {
+    return operandsLoadedErrorMessage ? (
+      <OperandsLoadedErrorAlert operandsLoadedErrorMessage={operandsLoadedErrorMessage} />
+    ) : (
+      (!operandsLoaded || (operandsLoaded && operands.length > 0)) && (
+        <span className="co-operator-uninstall__operands-section">
+          <h2>{t('olm~Operand instances')}</h2>
+          <OperandsTable
+            operands={operands}
+            loaded={operandsLoaded}
+            csvName={csvName}
+            cancel={cancel} // for breadcrumbs & cancel modal when clicking on operand links
+          />
+          <Checkbox
+            onChange={({ currentTarget }) => setDeleteOperands(currentTarget.checked)}
+            name="delete-all-operands"
+            label={t('olm~Delete all operand instances for this operator')}
+            checked={deleteOperands}
+          />
+        </span>
+      )
+    );
+  };
+
+  const OperandDeletionAlert = () =>
```

Comment:
No reason to make this a component.

```suggestion
  const operandDeletionAlert = operandDeletionErrors.length ? <OperandDeletionErrorAlert ... /> // etc
```